core/src/main/scala/org/apache/spark/sql/SnappyContext.scala

```diff
@@ -1273,3 +1282,32 @@ case class ExternalClusterMode(override val sc: SparkContext,
 
 class TableNotFoundException(message: String, cause: Option[Throwable] = None)
     extends AnalysisException(message) with Serializable
+
+/**
+ * Expression that returns the dsid of the server containing the row.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_() - Returns the dsid of the server containing the row.")
+case class DSID() extends LeafExpression {
+
+  override def nullable: Boolean = false
+
+  override def dataType: DataType = StringType
+
+  override val prettyName = "DSID"
+
+  override def eval(input: InternalRow): UTF8String = {
+    UTF8String.fromString(com.gemstone.gemfire.internal.cache.GemFireCacheImpl.getInstance()
+      .getDistributionManager().getId().getId());
+  }
+
+  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+
+    val idTerm = ctx.freshName("dsid")
+    ctx.addMutableState("UTF8String", idTerm, "")
+    ctx.addPartitionInitializationStatement(s"$idTerm = UTF8String.fromString(com.gemstone" +
```

Comment:
No not any specific reason i thought that initialization should go in init method